File: AgentApp/Backend/src/main/java/jvn/RentACar/model/RentRequest.java
package jvn.RentACar.model;

import jvn.RentACar.enumeration.RentRequestStatus;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.util.HashSet;
import java.util.Set;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class RentRequest {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Enumerated(EnumType.STRING)
    private RentRequestStatus rentRequestStatus;

    @ManyToOne(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    private Client client;

    @Column(nullable = false)
    private Double totalPrice;

    @OneToMany(mappedBy = "rentRequest", fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    private Set<RentInfo> rentInfos = new HashSet<RentInfo>();


}

File: AgentApp/Frontend/src/app/model/rentRequest.ts
import { RentInfo } from './rentInfo';
import { Client } from './client';

export class RentRequest {
    id: number;
    client: Client;
    rentInfos: RentInfo[];
    totalPrice: number;
    rentRequestStatus: string;
    constructor(client: Client, rentInfos: RentInfo[], totalPrice?: number, rentRequestStatus?: string, id?: number) {
        this.client = client;
        this.rentInfos = rentInfos;
        this.totalPrice = totalPrice;
        this.rentRequestStatus = rentRequestStatus;
        this.id = id;
    }

}